### PR TITLE
Watch all resources owned by AccountIAM CR

### PR DESCRIPTION
**Pull request by YCShen1010**
Issue: https://github.ibm.com/IBMPrivateCloud/roadmap/issues/64443

Ensure that whenever one of the owned resources (Deployment, Secret, ConfigMap ...) is modified or deleted, the controller will automatically reconcile the associated AccountIAM CR.

### CODE REVIEW

File: internal/controller/accountiam_controller.go [hunk 1]
```diff
@@ -897,5 +899,11 @@ func (r *AccountIAMReconciler) createOrUpdate(ctx context.Context, obj *unstruct
 func (r *AccountIAMReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&operatorv1alpha1.AccountIAM{}).
+		Owns(&appsv1.Deployment{}).
+		Owns(&corev1.Secret{}).
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.Service{}).
+		Owns(&corev1.ServiceAccount{}).
+		Owns(&routev1.Route{}).
```
Comment (bitscuit) [inline internal/controller/accountiam_controller.go:907]: the rbac resources are missing, right? Role and Rolebinding?

Comment (YCShen1010) [reply internal/controller/accountiam_controller.go:909]: Yes, I have added RoleBinding to be watched by the controller, but the Role keeps changing, which triggers reconciliation continuously. So, I temporarily commented out the resources that change during each reconciliation and need to figure out the cause

**Review by Daniel-Fan — APPROVED** [review]
/lgtm

**Review by Daniel-Fan — APPROVED** [review]
/lgtm

**Pull request merged**